linode/profile/framework_datasource.go

```diff
@@ -0,0 +1,153 @@
+package profile
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/linode/helper"
+)
+
+func NewDataSource() datasource.DataSource {
+	return &DataSource{}
+}
+
+type DataSource struct {
+	client *linodego.Client
+}
+
+func (data *DataSourceModel) parseProfile(ctx context.Context, profile *linodego.Profile) diag.Diagnostics {
+	data.Email = types.StringValue(profile.Email)
+	data.Timezone = types.StringValue(profile.Timezone)
+	data.EmailNotifications = types.BoolValue(profile.EmailNotifications)
+	data.Username = types.StringValue(profile.Username)
+	data.IPWhitelistEnabled = types.BoolValue(profile.IPWhitelistEnabled)
+	data.LishAuthMethod = types.StringValue(string(profile.LishAuthMethod))
+
+	authorized_keys, diag := types.ListValueFrom(ctx, types.StringType, profile.AuthorizedKeys)
+	if diag.HasError() {
+		return diag
+	}
+
+	data.AuthorizedKeys = authorized_keys
+
+	data.TwoFactorAuth = types.BoolValue(profile.TwoFactorAuth)
+	data.Restricted = types.BoolValue(profile.Restricted)
+
+	referrals, diag := flattenReferral(ctx, profile.Referrals)
+	if diag.HasError() {
+		return diag
```

Comment:
```suggestion
	referrals, diags := flattenReferral(ctx, profile.Referrals)
	if diags.HasError() {
		return diags
```

`diags` and `diag` actually imply different object type. I think it would be nice to rename it.

Reply:
Fixed!